cloudinit/osys/network.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2015 Cloudbase Solutions Srl
+# Copyright 2015 Canonical Ltd.
+# License TBD
+
+import abc
+import collections
+
+import six
+
+from cloudinit.osys import util
+
+
+__all__ = (
+    'BaseNetwork',
+    'BaseRoute',
+    'BaseRouteCollection',
+    'BaseInterface',
+)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseNetwork(object):
+    """Base network class for network related utilities."""
+
+    @abc.abstractmethod
+    def routes(self):
+        """Get an object representing the available routes."""
+
+    @abc.abstractmethod
+    def default_gateway(self):
+        """Get the default gateway, as a route object."""
+
+    @abc.abstractmethod
+    def interfaces(self):
+        """Get an object representing the available interfaces."""
+
+    @abc.abstractmethod
+    def hosts(self):
+        """Get an object representing the available hosts."""
+
+    @abc.abstractmethod
+    def set_hostname(self, hostname):
+        """Change the host name of the instance."""
+
+    @abc.abstractmethod
+    def set_timezone(self, timezone):
+        """Change the current timezone with the given timezone."""
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseRoute(object):
+    """Base class for routes."""
+
+    def __init__(self, destination, gateway, netmask,
+                 interface, metric,
+                 flags=None, refs=None, use=None, expire=None):
+        self.destination = destination
+        self.gateway = gateway
+        self.netmask = netmask
+        self.interface = interface
+        self.metric = metric
+        self.flags = flags
+        self.refs = refs
+        self.use = use
+        self.expire = expire
+
+    @abc.abstractproperty
+    def static(self):
+        """Check if this route is static."""
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseRouteCollection(collections.Sequence):
```

Comment:
So I'm still wondering, but do we need to prefix all these with 'Base'; is that adding any real value?